### ISSUE TITLE
Use GithHub discussions for feature requests, ideas and improvements

### ISSUE THREAD

**Issue by Jupeyy** [ddnet/ddnet#4001]
We have so many issues currently. Gets a bit messy

**Comment by edg-l** [comment]
I agree

**Comment by Jupeyy** [comment]
@Learath2 @def- if it's ok for you, i'd enable it, test it out, and if it doesn't work well we can still disable it again

**Comment by Jupeyy** [comment]
you can see it here:
https://github.com/ddnet/ddnet/discussions

**Comment by heinrich5991** [comment]
What's the criteria for moving stuff to discussions? What's the point of moving stuff to discussions? It seems to me that it's used for prioritizing?

For example https://github.com/ddnet/ddnet/issues/1966 was moved to discussions even though it seems like it focuses on abusable behavior of the DDNet server and even has two fix proposals. I think that should stay an issue? What do you think?

**Comment by heinrich5991** [comment]
It seems that you can't close discussions/keep track of which discussions are finished? E. g. https://github.com/ddnet/ddnet/discussions/4078 doesn't seem like a good fit for such an open ended discussion.

**Comment by edg-l** [comment]
I moved most discussions that seemed to want to introduce new features or change something that already exists, I had in mind keeping issues as issues things like bugs/security stuff/building issues.

As for #1966 i guess i didn't read it enough and moved it my bad. I think it can be converted back.

It's true that #4078 also doesn't seem like a good fit for discussions, I'll move it back.

**Comment by edg-l** [comment]
The main point of moving stuff to discussions is that we were getting tons of issues created for the sake of feature ideas, I think its good to have its own place for that and have the issues section focus on bugs, etc this way we can have a easy view at the quantity of real issues we have.

**Comment by Jupeyy** [comment]
I'm not so sure about #4078
Even now we still have around 300 issues.

And it seems this number tends to increase rather than decrease, so at some point we have to filter more aggressive, else "real" issues are lost.

We have alot of enhancement labeled issues, that are probably not really imeplemented any time soon.
We have issues that might not even be part of our code
#1921
#3909
#2757
and many more, that are never going to be fixed and just bloat the amount of issues

#4078 is definitelly more technical than a "feature / idea" but it's still more of a discussion than an actual issue.
So either close it as wontfix, discuss it somewhere else, or just open the pull request to it and discuss it inside the pr

**Comment by heinrich5991** [comment]
I currently don't see the improvement in using Github discussions.

The goal is to keep issues to stuff that will be accepted if PRed? Perhaps we could add a label for that?

**Comment by Jupeyy** [comment]
I think the idea makes sense to split it from issues completly, but you are right that there is no huge improvement, bcs most ppl still just open an issue.
Right now the issue count is already over 300 again, and its just too easy to loose track with so many issues ^^

**Comment by heinrich5991** [comment]
I'm in favor of disabling GitHub discussions again. We can add some kind of "actionable" and "unactionable" label that you can use to track which issues can be fixed and which can be ignored for the issue count. Currently, it looks like GitHub discussions is just a graveyard, no one comments over there.

**Comment by Jupeyy** [comment]
Labels aren't used either most of the time.. We should move all issues that are enhancements or feature request to the discussions more aggressively instead

**Comment by C0D3D3V** [comment]
We could just add issue templates, for auto labeling, and the first maintainer who reads it could make the lables more specific for what part the issue is.

I also do not like the discussion board very much, looks for me more like putting a issue in a corner to forget about it, because its not that important.

**Comment by edg-l** [comment]
I mean, lets be real, we have 340 issues and we only mostly touch the newer ones. The issues section is a graveyard too.

**Comment by edg-l** [comment]
We should have a bot to auto close issues after some time of inactivity and tag them, we can always re-open them. This will allow us to focus easier on the most recent and important issues. Since we can assume if a issue is inactive it isn't that important as of now.

**Comment by Jupeyy** [comment]
Yeah. For me it's also more about the ever increasing issue count.
Maybe someone has better ideas but generally I agree with ryo

**Comment by heinrich5991** [comment]
I don't see "decreasing the open issue count" as an advantage that outweighs the disadvantage of the split into two discussion venues.

> We should move all issues that are enhancements or feature request to the discussions more aggressively instead

Is there an advantage to doing that except for the "issue count" thing? I proposed a solution for the "actionable issues" that you can look at if you want to fix stuff.

**Comment by heinrich5991** [comment]
> We should have a bot to auto close issues after some time of inactivity and tag them, we can always re-open them.

No. This is a bad idea and will bury real issues that real users reported to us.

**Comment by C0D3D3V** [comment]
We could add some sort of voting system. If enough maintainers argue against the proposed new feature, we could close it instead of leaving it open and mark it as rejected. And if there are no arguments against it, or even maintainers are in favor of it, we mark it as an accepted feature request so that someone who is interested can implement it and make a PR. This can then remain open until there are arguments against it or it is implemented.

**Comment by Jupeyy** [comment]
So our situation is the following:
Nobody really using labels (i sometimes add them and learath and ryo too)
Many issues are system related bugs(vsync not working etc. just scroll up this discussions, i listed a few)
Discussion are just as dead, but at least they don't annoy me as developer to pick proper issues to fix

Heinrich wants actionable issues.
Imo wont work just as labels(in fact he proposed it as labels) or discussion page or whatever, simply bcs nobody cares just as now

Ryo wants a bot:
Can defs close real issues, but on the good site:
Creates more pressure on the devs / issue creators to fix stuff(bcs they get notified about closed issues and have to reopen) -> more discussion to dead issues & dead issues are just dead.
Defs a could work solution IMO

Codedev wants priorisation of featrues, e.g. by voting. Generally worth to think about IMO, but requires manually making an issue to feature request anyway, so basically not hugly different to discussion page, which also has a voting system.

And robyte correctly mentions that the discussion page needs more work from GitHub to be more useful generally

So nobody has a real solution and everyone is against everybody else's solution

nice

codedev also mentioned "issue templates"
I think as a first start we can add that and maybe demand a checkbox or smth if the issue is a feature request, bug etc. So some bot automatically adds labels?

**Comment by Jupeyy** [comment]
We can ofc also think about adding more maintainers, or maintainers specificly for caring about issues(if such ppl exists xd) from the community or the devs around here.

**Comment by heinrich5991** [comment]
> Discussion are just as dead, but at least they don't annoy me as developer to pick proper issues to fix

I'd like to move the discussions back to the issues tab. How about I label them with `discussions` or so that you can exclude them in your search for proper issues to fix?

**Comment by Jupeyy** [comment]
I really don't understand what fixes it for you but go ahead. The discussion will stay just as dead and we will have 400 open issues.

**Comment by heinrich5991** [comment]
I disabled the discussions tab. Existing discussions are lost until we find a way to migrate them back. Existing issues that were moved still exist, but lack newer comments and cannot be re-opened.

I find it irresponsible from GitHub to provide a way to move them one way, but not back. Still, I think the next time such a migration is done, the possibility of undoing it should be evaluated first.

**Comment by heinrich5991** [comment]
I like that they didn't stay as dead. E.g. I probably wouldn't even have learned about your comment here, if that had been in discussions.
